segmentation_models_pytorch/metrics/functional.py

```diff
@@ -260,14 +260,23 @@ def _compute_metric(
         tn = tn.sum()
         score = metric_fn(tp, fp, fn, tn, **metric_kwargs)
 
-    elif reduction == "macro" or reduction == "weighted":
+    elif reduction == "macro" :
         tp = tp.sum(0)
         fp = fp.sum(0)
         fn = fn.sum(0)
         tn = tn.sum(0)
         score = metric_fn(tp, fp, fn, tn, **metric_kwargs)
         score = _handle_zero_division(score, zero_division)
         score = (score * class_weights).mean()
+        
```

Comment:
What is the difference with "macro"?

Reply:
`sum()` versus `mean()` as far as I can see

Reply:
@JulienMaille Exactly.
sum() instead of mean() when reduction=="weighted", and class_weights are provided and not set to 1.

Reply:
line 254 normalizes the provided weights, so it has to sum to one:
`class_weights = class_weights / class_weights.sum()`
what do you think ?

Reply:
Yeah, that's right, thanks